R/mail.R

```diff
@@ -6,110 +6,249 @@
 #' @export
-convert_pipermail_to_mbox <- function(filelist) {
+refresh_pipermail <- function(mailing_list, start_year_month, save_folder_path) {
 
-  #at to @ replace function
-  pipermail_atreplacer <- function(string) {
+  # Create directory if it does not exist
+  if (!dir.exists(save_folder_path)) {
+    dir.create(save_folder_path, recursive = TRUE)
+  }
 
-    rstring <- sub(" at ", "@", string)
+  # Check if the folder is empty
+  files_in_folder <- list.files(save_folder_path, pattern = "kaiaulu_\\d{6}\\.mbox$")
+  if (length(files_in_folder) == 0) {
+    # If empty, download from start_year_month to the current month
+    end_year_month <- format(Sys.Date(), "%Y%m")
+    cat("Folder is empty. Downloading from", start_year_month, "to", end_year_month, "\n")
+    download_pipermail(mailing_list, start_year_month, end_year_month, save_folder_path)
+    return(NULL)
+  }
+  # If folder is not empty, find the most recent month
+  year_months <- gsub("kaiaulu_(\\d{6})\\.mbox$", "\\1", files_in_folder)
+  recent_month <- max(year_months)
+
+  # Delete the most recent file
+  recent_file <- file.path(save_folder_path, paste0("kaiaulu_", recent_month, ".mbox"))
+  if (file.exists(recent_file)) {
+    file.remove(recent_file)
+    cat("Deleted the most recent file:", recent_file, "\n")
+  }
 
-    return(rstring)
+  # Redownload from the most recent month to the current real-life month
+  end_year_month <- format(Sys.Date(), "%Y%m")
+  cat("Redownloading from", recent_month, "to", end_year_month, "\n")
+  download_pipermail(mailing_list, recent_month, end_year_month, save_folder_path)
+}
 
-  }
 
-  output <- "output.mbox"
+#' Process .gz files in a folder, unzip and convert them to .mbox
+#' Checks a folder for any .gz files, unzips them, and renames them
+#' to .mbox format. The original .gz files are deleted after unzipping. If a .mbox
+#' file with the same name already exists, it will be overwritten.
+#'
+#' @param folder_path The path to the folder containing both .gz and .mbox files.
+#' @return A list of the .mbox files that were created or updated.
+#' @export
+process_gz_to_mbox_in_folder <- function(folder_path) {
 
-  #Create mbox file and file connection
-  file.create(output)
-  fileConn <- file(output, "w+")
+  # Get the list of files in the folder
+  files <- list.files(folder_path, full.names = TRUE)
 
+  # Find .gz files
+  gz_files <- files[grepl("\\.gz$", files)]
 
-  #Read lines from downloaded files and write them to mbox file
-  for (filename in filelist[]){
+  # Check if there are no .gz files
+  if (length(gz_files) == 0) {
+    cat("This folder does not contain any .gz files.\n")
+    return(NULL)
+  }
 
-    #Open read connection
-    readCon <- file(filename, "r")
+  # Vector to store names of converted .mbox files
+  converted_mbox_files <- c()
 
-    data <- readLines(filename)
+  # Process .gz files
+  for (gz_file in gz_files) {
```

Comment:
Is the logic of converting gz files to mbox duplicated in download_pipermail? If so I guess it is ok just want to make sure.

Reply:
It is duplicated, which is why I was wondering if we need this function.